base_layer/core/src/chain_storage/memory_db/memory_db.rs

```diff
@@ -732,17 +715,61 @@ fn rewind_checkpoint_index(cp_count: usize, steps_back: usize) -> usize {
     }
 }
 
-// Calculate the total leaf node count upto a specified height.
-fn count_mmr_nodes_added(checkpoints: &MemDbVec<MerkleCheckPoint>, height: u64) -> Result<u32, ChainStorageError> {
-    let mut node_count: u32 = 0;
-    let last_index = min(checkpoints.len()?, (height + 1) as usize);
-    for cp_index in 0..last_index {
-        if let Some(cp) = checkpoints
-            .get(cp_index)
-            .map_err(|e| ChainStorageError::AccessError(format!("Checkpoint error: {}", e.to_string())))?
-        {
-            node_count += cp.nodes_added().len() as u32;
-        }
+/// Returns the accumulated node added count.
+///
+/// ## Panics
+///
+/// This will panic if the underlying checkpoint RwLock is poisoned
+fn fetch_last_mmr_node_added_count(checkpoints: &MemDbVec<MerkleCheckPoint>) -> u64 {
+    let cp_len = checkpoints.len().expect("MemDbVec RwLock is poisoned");
```

Comment:
I would have rather used the `.map_err(|e| ChainStorageError::AccessError(format!("Checkpoint error: {}", e.to_string())))?`and return a result instead of `expect` to keep the memory_db and lmdb_db version as similar as possible. Many tests only use the memory_db for testing.

Reply:
This means the new function for memorydb needs to return a result

Reply:
Ok, I see.